charts/core/src/Echart/useEchart.ts

```diff
@@ -0,0 +1,386 @@
+              zoomSelectionEvent[axis] = {
+                startValue: params.startValue,
+                endValue: params.endValue,
+              };
+              callback(zoomSelectionEvent);
+            } else if (params.batch) {
+              // Handle batch zoom (multiple axes)
+              params.batch.forEach((batchItem: any) => {
+                if (batchItem.dataZoomId?.includes('y')) {
+                  zoomSelectionEvent.yAxis = {
+                    startValue: batchItem.startValue,
+                    endValue: batchItem.endValue,
+                  };
+                } else {
+                  zoomSelectionEvent.xAxis = {
+                    startValue: batchItem.startValue,
+                    endValue: batchItem.endValue,
+                  };
+                }
+              });
+              callback(zoomSelectionEvent);
+            }
+          };
+
+          // Store the wrapper function so we can remove it later
+          activeHandlers.current.set(handlerKey, zoomHandler);
+          echartsInstance.on('datazoom', zoomHandler);
+          break;
+        }
+
+        default: {
+          activeHandlers.current.set(handlerKey, callback);
+          echartsInstance.on(action, callback);
+        }
+      }
+    }),
+    [echartsInstance],
+  );
+
+  useEffect(() => {
+    (async function setupChart() {
+      try {
+        setError(null);
+
+        await initializeEcharts();
+
+        if (container) {
+          const newChart = echartsCore.init(container);
+          newChart.setOption(options);
+          window.addEventListener('resize', resizeHandler);
+          setEchartsInstance(newChart);
+          setReady(true);
+        }
+      } catch (err) {
+        setReady(false);
+        console.error(err);
+        setError(
+          err instanceof Error
+            ? err
+            : new Error('Failed to initialize ECharts'),
+        );
+      }
+    })();
+
+    return () => {
+      if (echartsInstance) {
+        window.removeEventListener('resize', resizeHandler);
+        echartsInstance.dispose();
+      }
+    };
+  }, [container]);
+
+  useEffect(() => {
+    setOptions(currentOptions => {
+      const updatedOptions = {
+        ...currentOptions,
+        color: chartSeriesColors[theme],
+      };
+      setEchartOptions(updatedOptions);
+      return updatedOptions;
+    });
+  }, [theme]);
+
+  // Clean up all handlers when the echartsInstance changes
+  useEffect(() => {
+    return () => {
+      if (echartsInstance) {
+        // Remove all registered handlers
+        activeHandlers.current.forEach((handler, key) => {
+          const [action] = key.split('-');
+
+          if (action === 'zoomselect') {
+            echartsInstance.off('datazoom', handler);
+          } else {
+            echartsInstance.off(action, handler);
+          }
+        });
+        activeHandlers.current.clear();
+      }
+    };
+  }, [echartsInstance]);
```

Comment:
Good call. Because all of the disposal stuff happens in that useEffect, this should all be there as well. I'll move this.

Reply:
On second thought, since the instance is already being disposed, I'll get rid of this and just add `activeHandlers.current.clear();` to that cleanup callback.